Add explicit setup step for opening HTTP/HTTPS port on reolink devices

davidn opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

The Reolink integration requires HTTP/HTTPS ports to be enabled on devices before the it can be configured in Home Assistant. The current setup instructions do not guide users to do this, it is simply left as a note under troubleshooting.

This commit rephrases the installation instructions to ensure users enable HTTP/HTTPS ports.

Because I only have a Video Doorbell WiFi available to me to test, I do not know if this issue occurs with other doorbells. I have therefore phrased the extra installation step conditional on HTTP/HTTPS not yet being available.

As a Linux-only household, this is causing me some difficulty, and I wish I knew about the dependency on a Windows/Mac client before purchasing the device.  Therefore I have also added a note to the list of models indicating this requirement for this model.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

N/A

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Documentation**
  - Enhanced setup instructions for Reolink cameras, emphasizing the configuration of camera credentials, enabling HTTP/HTTPS ports via Windows or Mac Client, and integrating with Home Assistant.
  - Clarified steps for various connection methods and setup processes to improve user guidance.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

davidn committed:
Add explicit setup step for opening HTTP/HTTPS port on reolink devices

The Reolink integration requires HTTP/HTTPS ports to be enabled on devices before the it can be configured in Home Assistant. The current setup instructions do not guide users to do this, it is simply left as a note under troubleshooting.

This commit rephrases the installation instructions to ensure users enable HTTP/HTTPS ports.

Because I only have a Video Doorbell WiFi available to me to test, I do not know if this issue occurs with other doorbells. I have therefore phrased the extra installation step conditional on HTTP/HTTPS not yet being available.

As a Linux-only household, this is causing me some difficulty, and I wish I knew about the dependency on a Windows/Mac client before purchasing the device.  Therefore I have also added a note to the list of models indicating this requirement for this model.

davidn committed:
Fix typo s/Assitant/Assistant/

frenck commented on source/_integrations/reolink.markdown:
```diff
@@ -288,10 +288,12 @@ The following models have been tested and confirmed to work:
 - Reolink Duo Floodlight ([PoE](https://reolink.com/product/reolink-duo-floodlight-poe/) and [Wi-Fi](https://reolink.com/product/reolink-duo-floodlight-wifi/))
 - Reolink TrackMix ([PoE](https://reolink.com/product/reolink-trackmix-poe/) and [Wi-Fi](https://reolink.com/product/reolink-trackmix-wifi/))
-- Reolink Video Doorbell ([PoE](https://reolink.com/product/reolink-video-doorbell/) and [Wi-Fi](https://reolink.com/product/reolink-video-doorbell-wifi/))
+- Reolink Video Doorbell ([PoE](https://reolink.com/product/reolink-video-doorbell/) and [Wi-Fi](https://reolink.com/product/reolink-video-doorbell-wifi/)) **
```
I don't think this applies to this specific model alone (as a matter of fact, I'm sure :D )

davidn replied:
Ideally we'd specific exactly which models do and don't have this issue, but as I only have this device, and Reolink doesn't document it, I will remove this note.

starkillerOG replied:
That is really hard.
All camera's need to have at least one of the HTTP/HTTPS ports enabled, but depending on the firmware version and time at which the camera was bought (the firmware version it is shipped with) the HTTPS port may or may not be enabled by default.

The default has changed back and forth between firmware versions and I don't think Reolink has decided yet if they want to have it enabled or disabled as default.

frenck commented on source/_integrations/reolink.markdown:
```diff
@@ -288,10 +288,12 @@ The following models have been tested and confirmed to work:
 
 *These models are discontinued and not sold anymore, they will continue to work with Home Assistant.
 
+**These models do not have HTTP/HTTPS ports open by default. You must use the Windows or Mac Client to enable one of them before configuring in Home Assistant.
```
Let's remove this note here, and make it part of the instructions below instead.

davidn replied:
Removed.

davidn committed:
Remove note on specific models requiring HTTP/HTTPS ports opened

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -326,7 +324,7 @@ When your camera has a LAN port (most Wi-Fi cameras also have a LAN port):
       - After that you can disconnect the LAN cable and the camera will automatically switch to the Wi-Fi connection.
    5. Now set up the Reolink Home Assistant integration using the credentials you just specified.
```
```suggestion
```

davidn replied:
Applied this change.

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -339,6 +337,14 @@ Normally, the digits are printed directly under the QR code. Alternatively, you
 
+### 2. Enabling HTTP/HTTPS ports
+
+Test if you can access the cameara by its IP address in your browser. If you cannot, in the [windows or Mac](https://reolink.com/software-and-manual/) client ensure either HTTP and/or HTTPS ports are enabled under **Settings** > **Network** > **Advanced** > **Port Settings**. See [additional instructions](https://support.reolink.com/hc/en-us/articles/900004435763-How-to-Set-up-Reolink-Ports-Setti ...
```
```suggestion
Test if you can access the cameara by its IP address in your browser. If you cannot, ensure at least one of the HTTP/HTTPS ports are enabled in the [windows or Mac](https://reolink.com/software-and-manual/) client under **Settings** > **Network** > **Advanced** > **Port Settings**. See [additional instructions](https://support.reolink.com/hc/en-us/articles/900004435763-How-to-Set-up-Reolink-Ports-Settings-via-Reolink-Client-New-Client-) on the Reolink site.
```

davidn replied:
Applied this change.

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -347,7 +353,6 @@ Then power up the camera while pointing it at the QR code. It takes about a minu
 ## Troubleshooting
 
 - Older firmware versions do not expose the necessary information the integration needs to function. Ensure the camera is updated to the [latest firmware](https://reolink.com/download-center/) prior to setting up the integration. Note that Reolink auto update and check for update functions in the app/windows/web client often do not show the latest available firmware version. Therefore check the v ...
-- Ensure at least one of the HTTP/HTTPS ports is enabled in the [windows](https://reolink.com/software-and-manual/)/web client under **Settings** > **Network** > **Advanced** > **Port Settings**, see [additional instructions](https://support.reolink.com/hc/en-us/articles/900004435763-How-to-Set-up-Reolink-Ports-Settings-via-Reolink-Client-New-Client-) on the Reolink site.
```
Please keep this troubleshooting step, since the config flow inside homeassistant links to here.
Better to have it 2 times in the docs.

davidn replied:
Applied this change.

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -303,17 +303,15 @@ However, these cameras can work with this integration through an NVR in which th
+### 1. Intializing and configuring camera credentials.
 
-### Connecting Reolink via app/client
+There are several ways to achieve this:
```
```suggestion
A brand new Reolink camera first needs to be connected to the network and initialized. During initialization, the credentials for the camera need to be set.
There are several ways to achieve this:
```

davidn replied:
Applied this change.

starkillerOG replied:
thanks!

starkillerOG reviewed (commented):
Thanks for this PR, Indeed it would be good to emphasize the need to have at least one of the HTTP/HTTPS ports enabeld.

davidn committed:
Wording improvements to Reolink setup instructions

- Keep the HTTP/HTTPS mention in the troubleshooting steps
- Remove "configure integration" step from web set up instructions - they appear later in the instructions
- Keep intro text on setup instructions.

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -303,17 +303,16 @@ However, these cameras can work with this integration through an NVR in which th
 
-A brand new Reolink camera first needs to be connected to the network and initialized. During initialization, the credentials for the camera need to be set.
-There are several ways to achieve this.
+### 1. Intializing and configuring camera credentials.
```
AI is right, @davidn could you fix this typo?

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -339,6 +337,14 @@ Normally, the digits are printed directly under the QR code. Alternatively, you
 
+### 2. Enabling HTTP/HTTPS ports
+
+Test if you can access the cameara by its IP address in your browser. If you cannot, in the [windows or Mac](https://reolink.com/software-and-manual/) client ensure at least one of the HTTP/HTTPS ports are enabled under **Settings** > **Network** > **Advanced** > **Port Settings**. See [additional instructions](https://support.reolink.com/hc/en-us/articles/900004435763-How-to-Set-up-Reolink-Ports ...
```
AI is right again, I am impressed.
The suggestion to add "`http://<your-camera-ip>`" is also not a bad idea.

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -303,17 +303,16 @@ However, these cameras can work with this integration through an NVR in which th
 
-A brand new Reolink camera first needs to be connected to the network and initialized. During initialization, the credentials for the camera need to be set.
-There are several ways to achieve this.
+### 1. Intializing and configuring camera credentials.
```
```suggestion
### 1. Initializing and configuring camera credentials.
```

starkillerOG committed:
fix spelling

starkillerOG commented on source/_integrations/reolink.markdown:
```diff
@@ -339,6 +337,14 @@ Normally, the digits are printed directly under the QR code. Alternatively, you
 
+### 2. Enabling HTTP/HTTPS ports
+
+Test if you can access the cameara by its IP address in your browser. If you cannot, in the [windows or Mac](https://reolink.com/software-and-manual/) client ensure at least one of the HTTP/HTTPS ports are enabled under **Settings** > **Network** > **Advanced** > **Port Settings**. See [additional instructions](https://support.reolink.com/hc/en-us/articles/900004435763-How-to-Set-up-Reolink-Ports ...
```
```suggestion
Test if you can access the camera by its IP address in your browser `https://<your-camera-ip>`. If you cannot, in the [windows or Mac](https://reolink.com/software-and-manual/) client ensure at least one of the HTTP/HTTPS ports are enabled under **Settings** > **Network** > **Advanced** > **Port Settings**. See [additional instructions](https://support.reolink.com/hc/en-us/articles/900004435763-How-to-Set-up-Reolink-Ports-Settings-via-Reolink-Client-New-Client-) on the Reolink site.
```

starkillerOG committed:
Update source/_integrations/reolink.markdown

starkillerOG reviewed (approved):
LGTM
